### PR TITLE
Remove test directory from `initialize_project`

### PR DESCRIPTION
**Pull request by Datseris**
Well, it just so happens that every month I find a new reason to regret merging #43. No more I will live with such regrets, I am reverting it!

---

The motivation is simple: this "test" directory is clearly targeting a Julia package. I am not sure how many scientists write consistent checks for their actual projects. I am for sure not one of them. The advantage of not having a test directory will be great: hopefully people will not tend to think that DrWatson is about Julia packages.

### CODE REVIEW

**Comment by asinghvi17** [comment]
I have noticed that it's a bit hard to write tests of experimental code 😛

**Comment by jessebett** [comment]
Hey @Datseris, 

Sad to have seen this get unmerged. Writing unit tests for scientific code is good for making sure aspects of your inference pipeline haven't changed in ways that break your previous assumptions.

**Comment by tamasgal** [comment]
We had long discussions about this and yes, DrWatson creates a project folder and not a Julia package, so we decided to not confuse the package testing mechanism with it.

Another point is that usually one creates Packages to consolidate functionalities which are then documented and tested. A DrWatson protect is therefore more like a collection of composition of such packages so unit tests in that sense are out of the scope. We are more likely talking about high or even higher level tests and regression tests which needs to be individually tailored. 

That being said, nothing prevents you from adding your own test folders in whatever structures you like and you can add test procedures for all kinds of cases but we simply could not come up with a common structure which fits most cases and does not confuse people.

**Comment by Datseris** [comment]
Sorry for the dissatisfaction Jesse, but sometimes decisions have to be made, and this made DrWatson appear too much like a package generator or package development tool, which is not. I made the choice in favor of reducing confusion with the result of increasing dissatisfaction in a (hopefully small) percentage of users. You can't always win...

I just want to point out a technical detail: https://github.com/JuliaDynamics/DrWatson.jl/issues/67 

Simply put, you cannot use the command `pkg> test YourScientificProject`.
